pkg/apis/performanceprofile/v2/performanceprofile_validation_test.go

```diff
@@ -323,30 +371,183 @@ var _ = Describe("PerformanceProfile", func() {
 		})
 	})
 
+	Describe("The getNodesList helper function", func() {
+		It("should not return any errors when at least one node is detected", func() {
+			// Get client with one node to test this case
+			nodeSpecs := []NodeSpecifications{}
+			nodeSpecs = append(nodeSpecs, NodeSpecifications{architecture: amd64, cpuCapacity: 1000, name: "node"})
+			validatorClient = GetFakeValidatorClient(nodeSpecs)
+
+			// There should be a non-empty node list and no error present
+			nodes, err := profile.getNodesList()
+			Expect(err).To(BeNil())
+			Expect(nodes.Items).ToNot(BeEmpty())
+		})
+		It("should return an error when nothing is detected", func() {
+			// Get client with no nodes to test this case
+			nodeSpecs := []NodeSpecifications{}
+			validatorClient = GetFakeValidatorClient(nodeSpecs)
+
+			// There should be an empty node list and error present
+			nodes, err := profile.getNodesList()
+			Expect(err).ToNot(BeNil())
```

Comment:
General comment - I think it is good to always at least partially match the error string to ensure your unit test is hitting the "right" error path.

Reply:
Updated in latest commit